corporate/lib/stripe.py

```diff
@@ -2195,23 +2330,60 @@ def on_paid_plan(self) -> bool:
 
     @override
     def add_sponsorship_info_to_context(self, context: Dict[str, Any]) -> None:
-        # TBD
-        pass
+        context.update(
+            realm_org_type=self.remote_realm.org_type,
+            sorted_org_types=sorted(
+                (
+                    [org_type_name, org_type]
+                    for (org_type_name, org_type) in Realm.ORG_TYPES.items()
+                    if not org_type.get("hidden")
+                ),
+                key=self.sponsorship_org_type_key_helper,
+            ),
+        )
+
+    @override
+    def get_sponsorship_request_session_specific_context(
+        self,
+    ) -> SponsorshipRequestSessionSpecificContext:
+        return SponsorshipRequestSessionSpecificContext(
+            realm_user=None,
+            user_info=SponsorshipApplicantInfo(
+                name=self.remote_user_data.full_name,
+                email=self.remote_user_data.email,
+                # TODO: Set user_role when determining which set of users can access the page.
+                role="Remote realm administrator",
+            ),
+            # TODO: Check if this works on support page.
+            realm_string_id=self.remote_realm.host,
+        )
+
+    @override
+    def save_org_type_from_request_sponsorship_session(self, org_type: int) -> None:
+        if self.remote_realm.org_type != org_type:
+            self.remote_realm.org_type = org_type
+            self.remote_realm.save(update_fields=["org_type"])
 
 
 class RemoteServerBillingSession(BillingSession):  # nocoverage
     """Billing session for pre-8.0 servers that do not yet support
     creating RemoteRealm objects."""
 
     def __init__(
-        self, remote_server: RemoteZulipServer, support_staff: Optional[UserProfile] = None
+        self,
+        remote_server: RemoteZulipServer,
+        support_staff: Optional[UserProfile] = None,
+        user: Optional[UserDataForRemoteBilling] = None,
+        realm: Optional[RealmDataForAnalytics] = None,
```

Comment:
The realm and user fields would always be `None` for this session type; removed this.